File: ServerSide/BallPassSchedulePattern/Players_QrcodeReaderTypes/InitFromQrcodeReaderPlayerDir/InitFromQrcodeReaderPlayer.py
import os, sys
sys.path.append("../..")
from Players_CommonPlayers.SuperPlayerDir.SuperPlayer import SuperPlayer 

class InitFromQrcodeReaderPlayer(SuperPlayer):
    def __init__(self):
        super().__init__()  # スーパークラスの初期化メソッドを呼び出す
        self.my_name = None

    def return_my_name(self):
        return "InitFromQrcodeReaderPlayer"

    def main(self):
        """
        このメソッド実行直前に、
        このスーパークラスのメンバ変数
        one_time_world_instanceに、
        最新のworldインスタンスを代入しています。
        
        このメソッド終了後、
        メインで使用しているworld_instanceを
        このスーパークラスのメンバ変数
        one_time_world_instanceで上書きし、
        更新しています。
        
        → つまり、このメソッド内で
        self.one_time_world_instanceを上書きすると、
        その内容が、反映されます。
        """
        
        """ 初期化 """
        print(f"{return_my_name}が実行されました。")
        list2d = self.one_time_world_instance.columnSplitterConcatPlayer.new_list2d
        
        

        # データは自身のイニシャライザのメンバ変数で保持、
        # またはjsonなどに出力した場合、そのパスを保持
        self.one_time_world_instance.initFromQrcodeReaderPlayer = self

        return "Completed"

File: InstanceSide/BallPassSchedulePattern/Players_QrcodeReaderTypes/InitFromQrcodeReaderPlayerDir/InitFromQrcodeReaderPlayer.cs
using UdonSharp;
using UnityEngine;

public class InitFromQrcodeReaderPlayer : SuperPlayer
{
    [Header("Players (Attach in Unity Editor)")]
    public ColumnSplitterConcatPlayer columnSplitterConcatPlayer; // Unityエディタでアタッチ
    public InitFromQrcodeReaderPlayer initFromQrcodeReaderPlayer; // 自分自身を保持（Unityエディタで設定）

    void Start()
    {
        ResetPlayer(); // 初期化処理
    }

    public void ResetPlayer()
    {
        myName = "InitFromQrcodeReaderPlayer"; // プレイヤー名を設定
    }

    public override string ReturnMyName()
    {
        return "InitFromQrcodeReaderPlayer";
    }

    public override string ExecuteMain()
    {
        /*
         * このメソッド実行直前に、SuperPlayerのメンバ変数
         * one_time_world_instanceに最新のworldインスタンスが代入されていることを想定
         */

        // 初期化メッセージを表示
        Debug.Log($"{ReturnMyName()} が実行されました。");

        if (columnSplitterConcatPlayer == null)
        {
            Debug.LogError("ColumnSplitterConcatPlayer がアタッチされていません。Unityエディタで設定してください。");
            return "Error";
        }

        // ColumnSplitterConcatPlayer からデータを取得
        int[][] list2d = columnSplitterConcatPlayer.newList2D;

        if (list2d == null || list2d.Length == 0)
        {
            Debug.LogWarning("ColumnSplitterConcatPlayer の newList2D が初期化されていません。");
        }
        else
        {
            Debug.Log($"取得したデータの行数: {list2d.Length}");
        }

        // 自身を保持する
        initFromQrcodeReaderPlayer = this;

        Debug.Log("InitFromQrcodeReaderPlayer の処理が完了しました。");
        return "Completed";
    }
}
